### PR TITLE
chore(cz, deepsource, workflows): environment configuration

### PR DESCRIPTION
**Pull request by nezno4**
# hore(cz, deepsource, workflows): environment configuration
--

## What? ✌️

The following tools have been configured in the project:

- [Commitizen](https://commitizen-tools.github.io/commitizen/),
- [DeepSource](https://deepsource.com/),
- .gitignore,
- pom.xml,
- [README.md](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax),
- PULL_REQUEST_TEMPLATE.md --> [1](https://contribute.freecodecamp.org/how-to-open-a-pull-request/) [2](https://www.pullrequest.com/blog/writing-a-great-pull-request-description/),
- [JaCoCo](https://www.baeldung.com/jacoco),
- workflows:
  - first.yml ---> This workflow is responsible for test running script
  - first-secret.yml ---> This workflow is responsible for test running script with secret
  - bumpversion.yml ---> This workflow is responsible for automatically incrementing the application version and generating a changelog with each push to the master branch
  - deepsource.yml ---> This workflow includes steps to check the code, run tests, generate coverage reports, and analyze the code using DeepSource.
  -  branch-name-check.yml ---> This workflow is responsible for imposing a pattern for the branch name

## What Implement successful? 🤔

- [x] .cz.yaml
- [x] .deepsource.toml
- [x] .gitignore
- [x] pom.xml
- [x] README.md
- [x] PULL_REQUEST_TEMPLATE.md
- [x] \workflows\first.yml
- [x] \workflows\first-secret.yml
- [x] \workflows\deepsource.yml
- [x] \workflows\bumpversion.yml
- [x] \workflows\ranch-name-check.yml

issue: -

### CODE REVIEW

**Comment by nezno4** [comment]
> fill in the information in readme.md about the security measures used if you have time

ok I will add

File: .github/workflows/branch-name-check.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+      - name: Validate branch name
+        run: |
+          if [[ "$GITHUB_EVENT_NAME" == "pull_request" ]]; then
+            BRANCH_NAME=${GITHUB_HEAD_REF}
+          else
+            BRANCH_NAME=${GITHUB_REF##*/}
+          fi
+          
+          if [[ ! "$BRANCH_NAME" =~ ^(docs|feature|bugfix|hotfix|release|chore|refactor)/[a-z0-9]+(-[a-z0-9]+)* ]]; then
```
Comment (nezno4) [inline .github/workflows/branch-name-check.yml:26]: ### Prefixes for Branch
- **docs/:** Branches used for configuration and modifications of the project environment.
- **feature/:** Branches used for developing new features.
- **bugfix/:** Branches used for fixing bugs.
- **hotfix/:** Branches used for urgent fixes.
- **release/:** Branches used for preparing a release.
- **chore/:** Branches used for non-code tasks like dependency, docs updates.
- **refactor/:**  Branches used for refactoring code.

Comment (Nezno6) [reply .github/workflows/branch-name-check.yml:26]: Yes this ^(docs|feature|bugfix|hotfix|release|chore|refactor)/[a-z0-9]+(-[a-z0-9]+)* includes all options

**Review by Nezno6 — APPROVED** [review]
It is ok

**Pull request merged**